File: oldtests/kits/cpp/bot.cpp

#include "kit.hpp"
#include <string>

int main() {
  kit::Agent agent = kit::Agent();
  // initialize
  agent.initialize();

  while(true) {
    // play rock
    std::cout << 'R' << std::endl;
    // end turn
    agent.end_turn();
    // wait for updates
    agent.update();
  }
  
  return 0;
}

File: oldtests/kits/ts/bot.ts
import { Agent } from './kit';

// create a new agent
const agent = new Agent();

// first initialize the agent, and then proceed to go in a loop waiting for updates and running the AI
agent.initialize().then(async () => {
  // eslint-disable-next-line no-constant-condition
  while (true) {
    console.log('R');

    // now we end our turn
    agent.endTurn();
    // wait for update from match engine
    await agent.update();
  }
});
